packages/datetime2/src/components/date-range-picker3/dateRangePicker3.tsx

```diff
@@ -265,17 +265,35 @@ export class DateRangePicker3 extends DateFnsLocalizedComponent<DateRangePicker3
 
         const { value, time } = this.state;
         const newValue = DateUtils.getDateTime(
-            value[dateIndex] != null ? DateUtils.clone(value[dateIndex]!) : new Date(),
+            value[dateIndex] != null ? DateUtils.clone(value[dateIndex]!) : this.getDefaultDate(dateIndex),
             newTime,
         );
         const newDateRange: DateRange = [value[0], value[1]];
         newDateRange[dateIndex] = newValue;
         const newTimeRange: DateRange = [time[0], time[1]];
         newTimeRange[dateIndex] = newTime;
+
         this.props.onChange?.(newDateRange);
         this.setState({ value: newDateRange, time: newTimeRange });
     };
 
+    private getDefaultDate = (dateIndex: number) => {
```

Comment:
```suggestion
    // When a user sets the time value before choosing a date, we need to pick a date for
    // them to have something to store the time on.
    // The default depends on the value of the other date since there's an invariant
    // that the left/0 date is always less than the right/1 date.
    private getDefaultDate = (dateIndex: number) => {
```